packages/gatsby/src/redux/actions/add-page-dependency.ts

```diff
@@ -23,7 +23,7 @@ export const createPageDependency = ({
   if (
     nodeId &&
     componentDataDependencies.nodes.has(nodeId) &&
-    componentDataDependencies.nodes.get(nodeId).has(path)
+    componentDataDependencies.nodes.get(nodeId)!.has(path)
```

Comment:
If you change it to `?.` then it will work even if it is undefined

Reply:
I like this better because it doesn't have runtime implications. The whole point is that it tells TS that we _know_ this won't lead to a problem. If we change it to `?.` then at runtime we'll be doing `== null` checks which will never be true.